### PR TITLE
Fix delete method with explicit args in DatasetRequestAPI

### PR DESCRIPTION
**Pull request by shubham3121**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

- add a delete method for DatasetRequestAPI with explicit args
- add a delete asset to DatasetRequestAPI
- use del_asset to delete asset in the Dataset class

Fixes: https://github.com/OpenMined/PySyft/issues/6991

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by IonesioJunior — APPROVED** [review]
LGTM! Fantastic work as always @shubham3121!

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py [hunk 1]
```diff
@@ -123,6 +123,42 @@ def __init__(self, client: AbstractNodeClient):
...
+    ) -> None:
+
+        if not skip_checks:
+            pref = input(
+                f"You are about to delete asset with id: {asset_id} in the dataset with id `{dataset_id}`? 🚨 \n"
+                "All information related to this asset will be permanently deleted.\n"
+                "Please enter y/n to proceed: "
+            )
+            while pref != "y" and pref != "n":
+                pref = input("Invalid input '" + pref + "', please specify 'y' or 'n'.")
```
Comment (gmuraru) [inline packages/syft/src/syft/core/node/common/client_manager/dataset_api.py:154]: MAYBE: Use f-strings.

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py [hunk 2]
```diff
@@ -123,6 +123,42 @@ def __init__(self, client: AbstractNodeClient):
...
+            if pref == "n":
+                sys.stdout.write("Asset deletion is cancelled.")
```
Comment (gmuraru) [inline packages/syft/src/syft/core/node/common/client_manager/dataset_api.py:156]: Q: Why we use `sys.stdout.write` and not `print`?

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py [hunk 3]
```diff
@@ -178,7 +214,7 @@ def purge(self, skip_check: bool = False) -> None:
         if not skip_check:
             pref = input(
                 "You are about to delete all datasets ? 🚨 \n"
-                "All information will be permanantely deleted.\n"
+                "All information will be permanently deleted.\n"
```
Comment (gmuraru) [inline packages/syft/src/syft/core/node/common/client_manager/dataset_api.py:217]: Kudos: Nice!

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py [hunk 4]
```diff
@@ -486,8 +523,8 @@ def delete(self, name: str, skip_check: bool = False) -> bool:
                 sys.stdout.write("Asset deletion cancelled.")
                 return False
 
-        DatasetRequestAPI(self.client).delete(
-            dataset_id=self.id, bin_object_id=asset_id
+        DatasetRequestAPI(self.client).del_asset(
+            dataset_id=str(self.id), asset_id=asset_id, skip_checks=True
```
Comment (gmuraru) [inline packages/syft/src/syft/core/node/common/client_manager/dataset_api.py:527]: Q: Can't we simply pass `self.id` as a string for the `dataset_id`?

**Comment by gmuraru** [comment]
Looks great!

Do we have tests for this?

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py (hunk 2 above)
Comment (shubham3121) [reply packages/syft/src/syft/core/node/common/client_manager/dataset_api.py:156]: Just to be extra sure we will write on stdout, but we can replace it with simple print.

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py (hunk 4 above)
Comment (shubham3121) [reply packages/syft/src/syft/core/node/common/client_manager/dataset_api.py:527]: Just to by pass my py error as self.id type is not defined. But let me find a more concrete solution.

**Comment by shubham3121** [comment]
@gmuraru we have tests for datasets, seems like they were not added to tox and CI. I have added them now, should be reflected in the CI.

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py (hunk 4 above)
Comment (shubham3121) [reply packages/syft/src/syft/core/node/common/client_manager/dataset_api.py]: Fixed in https://github.com/OpenMined/PySyft/pull/7079/commits/c664423caee20d946a3d9adc0fc55121dbd5fcc7

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py (hunk 2 above)
Comment (shubham3121) [reply packages/syft/src/syft/core/node/common/client_manager/dataset_api.py:156]: Fixed in: https://github.com/OpenMined/PySyft/pull/7079/commits/c664423caee20d946a3d9adc0fc55121dbd5fcc7

File: packages/syft/src/syft/core/node/common/client_manager/dataset_api.py (hunk 1 above)
Comment (shubham3121) [reply packages/syft/src/syft/core/node/common/client_manager/dataset_api.py]: Fixed in: https://github.com/OpenMined/PySyft/pull/7079/commits/c664423caee20d946a3d9adc0fc55121dbd5fcc7

**Review by gmuraru — APPROVED** [review]
LGTM!

**Review by IonesioJunior — APPROVED** [review]

File: tox.ini [hunk 5]
```diff
@@ -189,7 +189,7 @@ setenv =
     EMULATION = {env:EMULATION:false}
     HAGRID_ART = false
     PYTHONIOENCODING = utf-8
-    PYTEST_MODULES = {env:PYTEST_MODULES:frontend network e2e security}
+    PYTEST_MODULES = {env:PYTEST_MODULES:frontend network e2e security redis}
```
Comment (rasswanth-s) [inline tox.ini:192]: ❤️

**Review by rasswanth-s — APPROVED** [review]
Marvelous Work @shubham3121  🥇  🎸 
It is great that, we got redis tests in the tests suite

**Pull request merged**